server/services/doc.js

```diff
@@ -0,0 +1,177 @@
+    if (!data) {
+      return res.status(404).send({ message: 'Document not found' });
+    }
+    if ((data && token.userRoleId === 1) || data.access === 'public') {
+      return res.status(200).send(data);
+    }
+    if (data.ownerId === token.userId) {
+      return res.status(200).send(data);
+    }
+    if (data && token.userRoleId === data.dataValues.ownerRoleId && data.access === 'role') {
+      return res.status(200).send(data);
+    }
+    return res.status(401).send({ message: 'Cannot Access document' });
+  });
+};
+
+/**
+ * Get created document
+ * @param {object} req
+ * @param {function} res // Response
+ * @returns {object} specified document.
+ */
+module.exports.getAllDocuments = (req, res) => {
+  const jwtcode = req.headers.authorization;
+  const token = jwt.verify(jwtcode, secret);
+  let ownerId;
+  let whereCondition = {};
+  if (req.params.username) {
+    User.find({
+      where: {
+        username: req.params.username
+      }
+    }).then((data) => {
+      ownerId = data.id;
+      whereCondition = { ownerId };
+    }
+    );
+  }
+  if (token.userRoleId === 1) {
+    Doc.findAll({ order: [['published', 'DESC']],
+      offset: req.params.offset,
+      limit: req.params.limit,
+      whereCondition
+    })
+        .then(data => res.status(200).send(data));
+  } else {
+    Doc.findAll({ order: [['published', 'DESC']],
+      limit: req.params.limit,
+      offset: req.params.offset,
+      where: {
+        $or: {
+          ownerId,
+          ownerRoleId: token.userRoleId,
+          access: 'public',
+          $and: {
+            access: 'private',
+            ownerId: token.userId
+          }
+        }
+      },
+      attributes: ['id', 'published', 'title', 'access', 'content', 'ownerId', 'ownerRoleId']
+    }).then(data => res.status(200).send(data)
+      );
+  }
+};
+
+module.exports.searchDocuments = (req, res) => {
+  const jwtcode = req.headers.authorization;
+  const token = jwt.verify(jwtcode, secret);
+  let query;
+  if (token.userRoleId === 1) {
+    query = { ownerRoleId: req.params.ownerRoleId };
+    if (req.params.date) {
+      query = {
+        ownerRoleId: req.params.ownerRoleId,
+        published: {
+          $like: `%${req.params.date}%`
+        }
+      };
+    }
+  } else {
+    // If the user is trying to search a different role level, return unauthorised;
+    if (parseInt(req.params.ownerRoleId, 10) !== token.userRoleId) {
+      return res.status(401).send({ message: 'Cannot Access document' });
+    }
+    query = {
+      $or: {
+        ownerRoleId: token.userRoleId,
+        access: 'public',
+        $and: {
+          access: 'private',
+          ownerId: token.userId
+        }
+      }
+    };
+    if (req.params.date) {
+      query = {
+        $or: {
+          ownerRoleId: token.userRoleId,
+          access: 'public',
+          $and: {
+            access: 'private',
+            ownerId: token.userId
+          }
+        },
+        published: {
```

Comment:
Since this is the only property (`published`) you're adding to the `query` object why not just use `query.published` instead of redeclaring `query` again